esphome/components/tem3200/tem3200.cpp

```diff
@@ -0,0 +1,154 @@
+#include "tem3200.h"
+#include "esphome/core/log.h"
+#include "esphome/core/helpers.h"
+#include "esphome/core/hal.h"
+
+namespace esphome {
+namespace tem3200 {
+
+static const char *const TAG = "tem3200";
+
+enum ErrorCode {
+  NONE = 0,
+  RESERVED = 1,
+  STALE = 2,
+  FAULT = 3,
+};
+
+void TEM3200Component::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up TEM3200...");
+
+  uint8_t status(NONE);
+  uint16_t temperature_raw(0);
+  uint16_t pressure_raw(0);
+
+  // startup device delay
+  delay(10);
+
+  i2c::ErrorCode err = this->read_(status, temperature_raw, pressure_raw);
+  if (err != i2c::ERROR_OK) {
+    ESP_LOGCONFIG(TAG, "    I2C Communication Failed...");
+    this->mark_failed();
+    return;
+  }
+
+  switch (status) {
+    case RESERVED:
+      ESP_LOGE(TAG, "Invalid RESERVED Device Status");
+      this->mark_failed();
+      return;
+    case FAULT:
+      ESP_LOGE(TAG, "FAULT condition in the SSC or sensing element");
+      this->mark_failed();
+      return;
+    case STALE:
+      ESP_LOGE(TAG, "STALE Data. Data has been fetched since last measurement cycle");
+      break;
+  }
+  ESP_LOGCONFIG(TAG, "    Success...");
+}
+
+void TEM3200Component::dump_config() {
+  ESP_LOGCONFIG(TAG, "TEM3200:");
+  LOG_I2C_DEVICE(this);
+  LOG_UPDATE_INTERVAL(this);
+  LOG_SENSOR("  ", "Temperature", this->temperature_sensor_);
```

Comment:
Maybe Raw Temperature?  (Raw Pressure)

Reply:
This *is* the actual temperature.

Would you like the pressure sensor and configuration renamed raw_pressure?

Reply:
yup, especially since one is raw and the other is not